### ISSUE TITLE
Access logs are not beeing purged PAYARA-2822

### ISSUE THREAD

**Issue by fabide** [payara/Payara#2703]
# Description #
----------

When using access logs in http service and set "Max File Count" the access logs are note beeing purged after max is reached. This is already an open issue in glassfish. I don't know how you handle open issues in glassfish. Please feel free to close this issue if it's already on your roadmap.
https://github.com/javaee/glassfish/issues/21526

## Expected Outcome

Set "Max File Count" 10 or server.http-service.access-log.max-history-files=10
Keep last 10 access logs and delete any older

## Current Outcome

Nothing happens to older access log files

## Steps to reproduce (Only for bug reports) 

1 -**  Start the domain

    ./asadmin start-domain payaradomain

2 -**  Configure an HTTP service in the admin console:

![02-05-_2018_16-49-16](https://user-images.githubusercontent.com/38913227/39530245-dade56da-4e28-11e8-90b5-1c55a3482b43.png)

## Environment ##

- **Payara Version**:  4.1.2.174 #badassfish (build 192)
- **Edition**:  Full
- **JDK Version**: Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
- **Operating System**: Debian GNU/Linux 9.4 (stretch)
- **Database**: MariaDB

**Comment by smillidge** [comment]
Can you try with 4.181 as this PR https://github.com/payara/Payara/pull/2250 may have fixed the issue

**Comment by z-mathe** [comment]
Hi @smillidge,
In our environments (Win64, 4.1.2.181, JDK 1.8u131) since the beginning with Max File Count = 30, the old files still exists (even from previous years, as long as it is not manually maintained), I just checked it.
```
<http-service access-logging-enabled="true">
        <access-log max-history-files="30" write-interval-seconds="10"></access-log> 
```
There are a lot of similar issues on the Internet because of the above phenomenon (since the glassfish version 3, not just under Payara), but no one has corrected it yet.
I finally wanted to write about this today, but after a short searching I see someone else doing it instead of me. Thanks for @fabide!
An old dream would come true once it would work. :)
Do you still need some information?
Thank you!

**Comment by fabide** [comment]
@z-mathe thanks for checking this out with newer version!
@smillidge do you need a test on linux as well?

**Comment by smillidge** [comment]
No don't need a test on Linux thanks for the offer.

**Comment by khasunuma** [comment]
I can reproduce the issue. 

Access Logging does not use `logging.properties` files. It's different from server.log or notification.log. It defines its logger and settings in `com.sun.enterprise.web.PEAccessLogValve` class but 'Max File Count' (or `com.sun.enterprise.server.logging.max_history_files` system property) is not read.

**Comment by z-mathe** [comment]
That's for sure?
https://github.com/payara/Payara/blob/master/appserver/web/web-glue/src/main/java/com/sun/enterprise/web/PEAccessLogValve.java

**Comment by smillidge** [comment]
Fixed by #2824
